Add IDE0160 and IDE0161. (#30034)
dotnet/docs: docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md

| Property                 | Value                                  |
| ------------------------ | -------------------------------------- |
| **Rule ID**              | IDE0160                                |
| **Title**                | Use block-scoped namespace             |
| **Category**             | Style                                  |
| **Subcategory**          | Language rules (namespace preferences) |
| **Applicable languages** | C#                                     |
| **Options**              | `csharp_style_namespace_declarations`  |

| Property                 | Value                                  |
| ------------------------ | -------------------------------------- |
| **Rule ID**              | IDE0161                                |
| **Title**                | Use file-scoped namespace              |
| **Category**             | Style                                  |
| **Subcategory**          | Language rules (namespace preferences) |
| **Applicable languages** | C#                                     |
| **Options**              | `csharp_style_namespace_declarations`  |

## Overview

> These rules concern how namespaces are declared. For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.

Author: As stated in https://github.com/dotnet/roslyn/issues/60546#issuecomment-1086854845, it's unclear why this option is even necessary. And what happens if you set it to `file_scoped` and enable IDE0160 (use block scoped namespaces)?

Reviewer: The option is used when creating a new file in Visual Studio to determine whether the new file uses file scoped or not. This is true even if both IDE0160 and IDE0161 are disabled.

| Property                 | Value                                  |
| ------------------------ | -------------------------------------- |
| **Rule ID**              | IDE0160                                |
| **Title**                | Use block-scoped namespace             |
| **Category**             | Style                                  |
| **Subcategory**          | Language rules (namespace preferences) |
| **Applicable languages** | C#                                     |
| **Options**              | `csharp_style_namespace_declarations`  |

| Property                 | Value                                  |
| ------------------------ | -------------------------------------- |
| **Rule ID**              | IDE0161                                |
| **Title**                | Use file-scoped namespace              |
| **Category**             | Style                                  |
| **Subcategory**          | Language rules (namespace preferences) |
| **Applicable languages** | C#                                     |
| **Options**              | `csharp_style_namespace_declarations`  |

## Overview

> These rules concern [how namespaces are declared](../../../csharp/language-reference/keywords/namespace.md). For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.

Reviewer 2: ```suggestion
These rules apply to [how namespaces are declared](../../../csharp/language-reference/keywords/namespace.md). For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.
```

Author: ```suggestion
These rules apply to [namespace declarations](../../../csharp/language-reference/keywords/namespace.md). For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.
```

## Options

> The option value specifies whether namespace declarations should be block scoped or file scoped. By default, namespace declarations are block scoped. This option is used by Visual Studio to determine how namespaces are declared when new code files are added to projects. Visual Studio honors the option value even if both `IDE0160` and `IDE0161` are disabled.

Reviewer 2: ```suggestion
The option value specifies whether namespace declarations should be block scoped or file scoped. By default, namespace declarations are block-scoped. This option is used by Visual Studio to determine how namespaces are declared when new code files are added to projects. Visual Studio honors the option value even if both `IDE0160` and `IDE0161` are disabled.
```
